app/models/configuration_script_source.rb

```diff
@@ -4,4 +4,10 @@ class ConfigurationScriptSource < ApplicationRecord
   belongs_to  :manager, :class_name => "ExtManagementSystem"
 
   virtual_total :total_payloads, :configuration_script_payloads
+
+  def self.class_from_request_data(data)
+    manager = ExtManagementSystem.find(data['manager_resource'])
```

Comment:
data['manager_resource'] is API payload dependent.  Let's send in the object instead, i.e. class_for_manager(manager)